### PR TITLE
fix(tabs): enables more nested override patterns

### PR DESCRIPTION
**Pull request by chasestarr**
#### Description

map children was previously not cloning the element, therefore losing some nested props. adds a warning to migrate from `TabPanel` component to `Tab`. 

#### Scope

- [x] Patch: Bug Fix

### CODE REVIEW

File: src/tabs/tabs.js [hunk 1]
```diff
@@ -34,18 +33,30 @@ export default class Tabs extends React.Component<TabsPropsT> {
...
         active: key === activeKey,
         disabled: disabled || child.props.disabled,
         $orientation: orientation,
-        overrides: child.props.overrides,
         onSelect: () => this.onChange({activeKey: key}),
-      };
-      return <Tab {...props}>{child.props.title}</Tab>;
+      });
+
+      // const key = child.key || String(index);
```
Comment (gergelyke) [inline src/tabs/tabs.js]: should we remove this?

File: src/tabs/tabs.js [hunk 2]
```diff
@@ -34,18 +33,30 @@ export default class Tabs extends React.Component<TabsPropsT> {
     // eslint-disable-next-line flowtype/no-weak-types
     const tabs = React.Children.map(children, (child: any, index) => {
       if (!child) return;
+
       const key = child.key || String(index);
-      const props = {
-        key,
+      return React.cloneElement(child, {
+        key: key,
```
Comment (gergelyke) [inline src/tabs/tabs.js]: can we keep this simply `key,`?

File: src/tabs/tabs.js (hunk 1 above)
Comment (chasestarr) [reply src/tabs/tabs.js]: fixed thank you

File: src/tabs/tabs.js (hunk 2 above)
Comment (chasestarr) [reply src/tabs/tabs.js]:
```suggestion
        key,
```

**Pull request merged**